rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -52,7 +53,8 @@ trait Reduce[M[_]] {
 
 class DebruijnInterpreter[M[_], F[_]](
     tuplespaceAlg: Tuplespace[M],
-    urnMap: Map[String, Par]
+    urnMap: Map[String, Par],
+    deployParametersRef: Ref[M, DeployParameters]
```

Comment:
I'd love to avoid having yet another ref shared. Ideally, we'd create the Reduce instance just before interpretation, and pass the deploy parameters directly. We should be able to reuse the tuplespace between the interpreters. I'll open a ticket to clean that up.